src/resource_clients/request_queue.ts

```diff
@@ -103,33 +108,65 @@ export class RequestQueueClient extends ResourceClient {
         return cast(parseDateFields(pluckData(response.data)));
     }
 
+    /**
+     * Writes requests to request queue in batch.
+     * THIS METHOD IS EXPERIMENTAL AND NOT INTENDED FOR PRODUCTION USE.
+     *
+     * @private
+     * @experimental
+     */
+    protected async _batchAddRequests(
+        requests: Omit<RequestQueueClientRequestSchema, 'id'>[],
+        options: RequestQueueClientAddRequestOptions = {},
+    ): Promise<RequestQueueClientBatchRequestsOperationResult> {
+        ow(requests, ow.array.ofType(ow.object.partialShape({
+            id: ow.undefined,
+        })).minLength(1).maxLength(MAX_REQUESTS_PER_BATCH_OPERATION));
+        ow(options, ow.object.exactShape({
+            forefront: ow.optional.boolean,
+        }));
+
+        const { data } = await this.httpClient.call({
+            url: this._url('requests/batch'),
+            method: 'POST',
+            timeout: this.timeoutMillis,
+            data: requests,
+            params: this._params({
+                forefront: options.forefront,
+                clientKey: this.clientKey,
+            }),
+        });
+
+        return cast(parseDateFields(pluckData(data)));
+    }
+
     protected async _batchAddRequestsWithRetries(
         requests: Omit<RequestQueueClientRequestSchema, 'id'>[],
-        options: RequestQueueClientBatchAddRequestOptions = {},
-    ): Promise<RequestQueueClientBatchAddRequestsResult> {
+        options: RequestQueueClientBatchAddRequestWithRetriesOptions = {},
+    ): Promise<RequestQueueClientBatchRequestsOperationResult> {
+        const {
+            maxUnprocessedRequestsRetries = this.httpClient.maxRetries,
+            forefront,
+        } = options;
         // Keep track of the requests that remain to be processed (in parameter format)
         let remainingRequests = requests;
         // Keep track of the requests that have been processed (in api format)
         const processedRequests: ProcessedRequest[] = [];
         // The requests we have not been able to process in the last call
         // ie. those we have not been able to process at all
         let unprocessedRequests: UnprocessedRequest[] = [];
-        // TODO: Do not use `maxRetries` from http client, but rather a separate value
-        for (let i = 0; i < 1 + this.httpClient.maxRetries; i++) {
+        for (let attempt = 0; attempt < 1 + maxUnprocessedRequestsRetries; attempt++) {
```

Comment:
This is something we should discuss perhaps this value should be much larger not just 8. As in the majority of cases, you want it to keep retrying as long as it's needed although it may take 20mins.

Or we could override this when used from SDK.

CC @B4nan @mnmkng

Reply:
I think in terms of making updates to the queue, we prefer fast info. Some of the most annoying bugs in the SDK lately were caused by request queue getting stuck for minutes. It's one of the causes of the 0 concurrency bug. Anything more than 30 secs is a bad idea imo.

Reply:
> Anything more than 30 secs is a bad idea imo.

Timeout probably makes sense but 30 secs is too optimistic, as this method should be used to pass thousands of requests into the queue. From tests of batch insert, it looks like we can insert 1k requests per sec(of course in depends on how heavily the request queues are used at the moment.).

Reply:
Hmm, I'm not entirely sure about it, but I guess I wouldn't want to push one million requests with a single function call that can potentially last 1000 seconds. Maybe we should limit the batch size to 5000 or something? We can always increase the limit if we find ways to make this faster.

Reply:
cc @B4nan ad above

Reply:
We do override the timeout for request queue client to 30s in SDK (and retry 3 times ourselves), we actually believe that waiting for longer was one of the things behind the 0 concurrency bug. Agreed it makes sense for us to do things in smaller batches like 1000 requests and not wait for the whole import (cc @vladfrangu we should revisit this). But that can be already done, its up to us how many requests we pass to the method, right? I don't mind if the client itself allows unlimited amount, as I am in the control when I use it. We can await the first batch of 1000 and schedule the rest to run in background (but I guess serially so we don't put too much pressure on the API).

Reply:
@B4nan Yep, you can pass how many requests you want to and maxParallel option is set up to optimal values which come from testing, but you can change it if you see some issues.